sktime/forecasting/compose/_networkpipeline.py

```diff
@@ -0,0 +1,269 @@
+        """
+
+        returned_arguments_kwarg = arguments.copy()
+        if arguments is None:
+            return arguments
+        for argument_name, argument_value in returned_arguments_kwarg.items():
+            if argument_value == "original_X":
+                returned_arguments_kwarg[argument_name] = self._X
+                continue
+            if argument_value == "original_y":
+                returned_arguments_kwarg[argument_name] = self._y
+                continue
+            if argument_value == "original_fh":
+                returned_arguments_kwarg[argument_name] = self._fh
+                continue
+
+            if type(argument_value) == list:
+                out = []
+                for list_val in argument_value:
+                    returned_step_value = self._check_steps_for_values(list_val)
+                    if returned_step_value is not None:
+                        out.append(returned_step_value)
+                returned_arguments_kwarg[argument_name] = out
+                continue
+            # go through all steps and look for returned values
+            returned_step_value = self._check_steps_for_values(argument_value)
+            if returned_step_value is not None:
+                returned_arguments_kwarg[argument_name] = returned_step_value
+
+        return returned_arguments_kwarg
+
+    def fit(self, y, X=None, fh=None):
+        self._set_y_X(y, X)
+        self._set_fh(fh)
+        for name, alg, arguments in self._iter():
+            processed_arguments = {}
+
+            if "fit" in arguments and arguments["fit"] is None:
+                # this step must be skipped
+                continue
+            elif "fit" in arguments:
+                processed_arguments = self._process_arguments(arguments["fit"])
+            else:
+                processed_arguments = self._process_arguments(arguments)
+
+            if "y" in processed_arguments:
+                processed_arguments["y"] = check_y(processed_arguments["y"])
+            if "Z" in processed_arguments:
+                processed_arguments["Z"] = check_y(processed_arguments["Z"])
+            if "X" in processed_arguments and processed_arguments["X"] is not None:
+                processed_arguments["X"] = check_X(processed_arguments["X"])
+            if "X" in processed_arguments and processed_arguments["X"] is None:
+                del processed_arguments["X"]
+            # Transformers are instances of BaseTransformer and BaseEstimator
+            # Estimators are only instances of BaseEstimator
+            if hasattr(alg, "fit_transform"):
+                out = alg.fit_transform(**processed_arguments)
+                self._step_results[name] = out
+                self._is_fitted = True
+            processed_arguments["fh"] = self._fh
+            # estimators have fit and predict methods
+            if hasattr(alg, "fit") and hasattr(alg, "predict"):
+                alg.fit(**processed_arguments)
```

Comment:
Are you cloning the algorithm somewhere? Otherwise you will update the object in the list which may explain the `check_fit_does_not_overwrite_hyper_params` error.

Reply:
thanks, this fixed the error.